### PR TITLE
Fix to Skip Buffer Iteration Performance Fix 1205

### PR DESCRIPTION
**Pull request by kishorechavali**
### Summary:

Added Fix for differentiating Buffer from rest of object types to avoid iterating over entries in Buffer in Schema getTypePaths

### GitHub linked issue: https://github.com/dynamoose/dynamoose/issues/1205

#---
Closes

### Type (select 1):
- [x] Bug fix
- [ ] Feature implementation
- [ ] Documentation improvement
- [ ] Testing improvement

- [ ] Test added to report bug (GitHub issue #--- @---)
- [ ] Something not listed here

### Is this a breaking change? (select 1):
- [ ] 🚨 YES 🚨
- [x] No
- [ ] I'm not sure

### Is this ready to be merged into Dynamoose? (select 1):
- [x] Yes
- [ ] No

### Are all the tests currently passing on this PR? (select 1):
- [x] Yes
- [ ] No

### Other:
- [x] I have read through and followed the Contributing Guidelines
- [x] I have searched through the GitHub pull requests to ensure this PR has not already been submitted
- [x] I have updated the Dynamoose documentation (if required) given the changes I made
- [x] I have added/updated the Dynamoose test cases (if required) given the changes I made
- [x] I have ensured the following commands are successful from the root of the project directory
  - [x] `npm test`
  - [x] `npm run lint`
- [x] I agree that all changes made in this pull request may be distributed and are made available in accordance with the [Dynamoose license](https://github.com/dynamoose/dynamoose/blob/master/LICENSE)
- [x] All of my commits and commit messages are detailed, explain what changes were made, and are easy to follow and understand
- [x] I have filled out all fields above

### CODE REVIEW

**Review by fishcharlie — CHANGES_REQUESTED** [review]
You checked the `I have added/updated the Dynamoose test cases (if required) given the changes I made` box on the PR template. But I don't see any test cases added. Am I missing something?

**Comment by kishorechavali** [comment]
> You checked the `I have added/updated the Dynamoose test cases (if required) given the changes I made` box on the PR template. But I don't see any test cases added. Am I missing something?

Well since the code was only one line and i reviewed that there were no existing test cases for Buffer for method getTypePaths , i went by if required part and marked it as yes , should i update it to No ?
Also all existing test cases passed . Please take a look at my code change and let me know what testcase you have in mind or do you think it will be breaking change .

**Comment by fishcharlie** [comment]
> Well since the code was only one line and i reviewed that there were no existing test cases for Buffer for method getTypePaths , i went by if required part and marked it as yes , should i update it to No ?
> Also all existing test cases passed . Please take a look at my code change and let me know what testcase you have in mind or do you think it will be breaking change .

@kishorechavali Yeah it needs test cases. What is to prevent this from having a regression later on? The only changes that don't require new or updated test cases are documentation and process changes.

I'm not quite sure exactly what this test case(s) would look like, but it for sure needs something to ensure it doesn't have a regression later on.

**Comment by fishcharlie** [comment]
I'd be happy to help with figuring out and writing some tests for this, but I'm not sure when I'll have time to get around to doing that.

**Comment by kishorechavali** [comment]
Challenge with th

> I'd be happy to help with figuring out and writing some tests for this, but I'm not sure when I'll have time to get around to doing that.

This is a bug fix esp for differentiating Buffer from Object to skip recursive traversal for Buffer objects , because of this it is tricky to target a specific testcase . The issue gets compounded especially when we have huge buffer , ex in our case we were gziping documents . If Buffer size is small we dont notice the issue. I am open on collaborating to come up with testcase

File: lib/Schema.ts [hunk 1]
```diff
@@ -421,7 +421,8 @@ export class Schema {
 			}
 			const {typeDetails, matchedTypeDetailsIndex, matchedTypeDetailsIndexes} = typeCheckResult;
 			const hasMultipleTypes = Array.isArray(typeDetails);
-			const isObject = typeof value === "object" && value !== null;
+			// Added Fix for differentiating Buffer from rest of object types to avoid iterating over entries in Buffer
```
Comment (fishcharlie) [inline lib/Schema.ts:424]:
```suggestion
```

Normally I'm a huge fan of code comments. But in this case the code is pretty self explanatory.

**Review by fishcharlie — CHANGES_REQUESTED** [review]
One comment, plus some lint errors, then I'll do a final review.

File: lib/Schema.ts (hunk 1 above)
Comment (fishcharlie) [reply lib/Schema.ts]: Fixed in 4bb73de

**Review by fishcharlie — APPROVED** [review]

**Pull request merged**